### DISCUSSION TITLE
PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted

### DISCUSSION THREAD

**Discussion by palansher** [composer-unused/composer-unused#423] [Q&A]
Good day!

I get memory exhausted error on memory_limit = 256 Mb on relative small project.

I increased till 512M and got save error.
How mush memory composer-unused needs?

14:17 $ vendor/bin/composer-unused

`PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 536870920 bytes) in /home/vladp/dev/gtdev/vendor/composer-unused/symbol-parser/src/Parser/PHP/Strategy/AnnotationStrategy.php on line 165`

composer-unused 0.8.2
PHP 8.1
Ubuntu 20.04

Thank you!

**Comment by icanhazstring** [comment]
Hi. Currently, this is a known issue of the annotations parsing.
https://github.com/composer-unused/symbol-parser/issues/66

My first thought is that this might be some kind of endless loop, just crashing.
I still have to figure out what is happening there.

If you have a small project. Is it possible that I can debug that for myself?
So I might find the issue there because reproducing it is kind of hard. :)

**Reply by palansher** [reply]
Dear Andreas, 
This is a business legacy project with many security holes, 180 000 lines of code. And my employer surely will not agree with sharing.
Really sorry.

**Reply by icanhazstring** [reply]
Totally understand 😉
